test/integration/framework/controlplane_utils.go

```diff
@@ -223,6 +224,11 @@ func startAPIServerOrDie(controlPlaneConfig *controlplane.Config, incomingServer
 	if controlPlaneConfig.ExtraConfig.ServiceIPRange.IP == nil {
 		controlPlaneConfig.ExtraConfig.ServiceIPRange = net.IPNet{IP: netutils.ParseIPSloppy("10.0.0.0"), Mask: net.CIDRMask(24, 32)}
 	}
+
+	if controlPlaneConfig.ExtraConfig.EndpointReconcilerType == "" {
```

Comment:
q: in createEndpointReconciler you fixed the default to use the lease reconciler but what's the reason for changing the integration tests to use "None"? Do none of the existing tests rely on the default Kubernetes service?

Reply:
yeah, this seems questionable... does the default of `lease` not work for integration tests?

Reply:
I was testing this, this commits is going away, I've found the root cause and I'm working with Abu on it
https://github.com/kubernetes/kubernetes/pull/108033

Reply:
the apiserver shutdown is not respecting the preshutdown hooks and racing with the reconcilers

Reply:
PTAL now, I've updated the PR, there were many things here :upside_down_face: